pkg/k8s/labels.go

```diff
@@ -397,6 +397,14 @@ const (
 	// services.
 	DefaultExportedServiceSelector = SvcMirrorPrefix + "/exported"
 
+	// GlobalServiceNameLabel returns whether or not the name is global or should be localized to
+	// each cluster it is mirrored on, containing the remote mirror name
+	GlobalServiceNameLabel = SvcMirrorPrefix + "/global-service-name"
```

Comment:
I like this approach. I do have some questions though:

* What's your opinion on having a boolean based value as opposed to being able to specify the name for the global service? On the one hand, specifying the names ourselves means we have more flexibility. OTOH, having a boolean value here (similar to `mirror.linkerd.io/exported`) means we can have stronger assumptions, in the future if we want to expand on this, it's trivial to get the name of the global service (since it'll be conventional). As someone who would interact with this, what's your opinion? Hope the question makes sense.

* We'll end up using the annotation value as the name for our global headless mirror. Should we enforce that the base name here matches that of the exported service? We should definitely document what this annotation will do, but even so, I think it could be a footgun in certain circumstances.